src/libmongoc/src/mongoc/mongoc-handshake.c

```diff
@@ -351,39 +352,31 @@ _get_system_info (mongoc_handshake_t *handshake)
    handshake->os_architecture = _get_os_architecture ();
 }
 
-static void
-_free_system_info (mongoc_handshake_t *handshake)
-{
-   bson_free (handshake->os_type);
-   bson_free (handshake->os_name);
-   bson_free (handshake->os_version);
-   bson_free (handshake->os_architecture);
-}
-
 static void
 _get_driver_info (mongoc_handshake_t *handshake)
 {
    handshake->driver_name = bson_strndup ("mongoc", HANDSHAKE_DRIVER_NAME_MAX);
    handshake->driver_version = bson_strndup (MONGOC_VERSION_S, HANDSHAKE_DRIVER_VERSION_MAX);
 }
 
-static void
-_free_driver_info (mongoc_handshake_t *handshake)
-{
-   bson_free (handshake->driver_name);
-   bson_free (handshake->driver_version);
-}
-
 static void
 _set_platform_string (mongoc_handshake_t *handshake)
 {
    handshake->platform = bson_strdup ("");
 }
 
 static void
-_free_env_info (mongoc_handshake_t *handshake)
+_get_container_info (mongoc_handshake_t *handshake)
 {
-   bson_free (handshake->env_region);
+   char *kubernetes_env = _mongoc_getenv ("KUBERNETES_SERVICE_HOST");
+   handshake->kubernetes = kubernetes_env;
+
+   handshake->docker = false;
+#ifdef _POSIX_VERSION
+   handshake->docker = (access ("/.dockerenv", F_OK) == 0);
+#endif
```

Comment:
Is a separate check needed if the Docker container is running Windows? Example:

```c
#ifdef _WIN32
   handshake->docker = (_access_s ("C:\\.dockerenv", 0) == 0);
#else
   handshake->docker = (access ("/.dockerenv", F_OK) == 0);
#endif
```

On non-Windows, I expect `access` can be assumed to be available since [POSIX features are enabled](https://github.com/mongodb/mongo-c-driver/blob/1b0e008d3aab07431827433c783f92680cb0cffe/CMakeLists.txt#L370).